File: src/lib/paths.ts
// Path utilities — expands home dir shortcuts and derives profile directory paths.

import path from "path";
import os from "os";
import { GitRc } from "./types.js";

export function resolvePath(p: string): string {
  const home = os.homedir();
  // handles both Unix (~) and Windows (%USERPROFILE%) conventions
  return p.replace(/^~/, home).replace(/%USERPROFILE%/g, home);
}

export function resolveProfilesDir(config: GitRc): string {
  return resolvePath(config.profilesDir);
}

export function getProfileDir(config: GitRc, profileName: string): string {
  return path.join(resolveProfilesDir(config), profileName);
}

File: src/util/paths.rs
use crate::state::config::GitRc;

pub fn resolve_path(p: &str) -> String {
    let home = dirs::home_dir()
        .map(|h| h.to_string_lossy().to_string())
        .unwrap_or_default();
    p.replace('~', &home).replace("%USERPROFILE%", &home)
}

pub fn resolve_profiles_dir(config: &GitRc) -> String {
    resolve_path(&config.profiles_dir)
}

pub fn get_profile_dir(config: &GitRc, profile_name: &str) -> String {
    let base = resolve_profiles_dir(config);
    format!("{base}/{profile_name}")
}
